### PR TITLE
Migrate to new iex cloud apis

### PR DESCRIPTION
**Pull request by bingxie**
Because

> On June 1, 2019, IEX API will sunset all non-IEX data. IEX Cloud, a non-Exchange platform, will continue to provide access to third-party data sources.
> see here: https://iextrading.com/developer/

I made those change to migrate to the new IEX cloud APIs.
Which includes:

1. Use the new IEX cloud URL and token
2. Change the crypto quote API 
3. Update several resources: dividends, earnings, key_stats 
4. Update all of the fixtures
5. Update README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +59,10 @@ Run `bundle install`.
 
 ## Usage
 
+### Set API token
+
+Before start to use this gem, set `IEX_CLOUD_SECRET_TOKEN` environment variable to your IEX cloud API token.
```
Comment (dblock) [inline README.md]: `To use this gem set ...`

Comment (dblock) [reply README.md]: Maybe add a link to where/how to obtain this token here?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -8,3 +8,5 @@
 Dir[File.join(File.dirname(__FILE__), 'support', '**/*.rb')].each do |file|
   require file
 end
+
+ENV['IEX_CLOUD_SECRET_TOKEN'] = 'test-iex-cloud-secret-token'
```
Comment (dblock) [inline spec/spec_helper.rb]: Make this `ENV[...] ||=` 

If one needs to record a new VCR I can then can do it without modifying code with `IEX_CLOUD_SECRET_TOKEN=... rspec spec/...`.

**Comment by bingxie** [comment]
Hello @dblock 
Thanks for you review. According to you feedback I have some questions: 
1.  They will sunset the old api on June 1, 2019, so do I need to mention this on README?
2.  Where should I put those UPGRADING description?
3. Should we bump the version to 1.0.0? IEX cloud sent me message they will have new version(version 1) next week.
4. I looked the document again, they only mentioned to put the token as parameter.

**Comment by dblock** [comment]
> 1. They will sunset the old api on June 1, 2019, so do I need to mention this on README?

Personally I had no idea they would, so maybe say something to that effect. NBD

> 2. Where should I put those UPGRADING description?

https://github.com/dblock/iex-ruby-client/blob/master/UPGRADING.md

> 3. Should we bump the version to 1.0.0? IEX cloud sent me message they will have new version(version 1) next week.

Yes, sounds good, you can do this as part of the PR.

> 4. I looked the document again, they only mentioned to put the token as parameter.

Maybe we shouldn't call it "secret token", just "IEX_API_TOKEN"? Where did you get the secret part from?

**Comment by dblock** [comment]
I've finally found the time to read https://iexcloud.io/docs/api/#authentication ;)

It looks like the token that you're trying to supply here is a _publishable_ token. So I would call this `IEX_API_PUBLISHABLE_TOKEN`. 

The API supports other workflows with a secret token, and we can defer implementing support for new APIs that take that for later. Once this is merged let's open ourselves some issues, including sandbox support and such.

**Comment by investment-radar** [comment]
@dblock I updated the token ENV name, and update to version 1 URI. Currently I am still working on the UPGRADING.md.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
-### 0.4.5 (Next)
+### 1.0.0 (Next)
```
Comment (dblock) [inline CHANGELOG.md]: Also bump it in https://github.com/dblock/iex-ruby-client/blob/master/lib/iex/version.rb

**Comment by dblock** [comment]
Looks good so far! Thanks for hanging in there.

**Comment by dblock** [comment]
Let's finish this @bingxie ? Need help?

**Comment by bingxie** [comment]
@dblock Updated the UPGRADING.md,  can you have another look? Thanks.

**Comment by dblock** [comment]
Merged it, thank you. 

Would you like help co-maintain the gem? If you do, shoot me an email to dblock at dblock dot org with your rubygems account. Maybe you can make the next release?

**Comment by dblock** [comment]
@bingxie Also please take a look at https://github.com/dblock/iex-ruby-client/pull/46, I am making the client configurable to enable #44, remove the need for an environment variable, etc.

**Pull request merged**